Azure/ARO-RP
# Network acl fix for storageaccounts on Create or AdminUpdate

### Which issue this PR addresses:

Fixes [ARO-4450](https://issues.redhat.com/browse/ARO-4450)

### What this PR does / why we need it:

to properly use `azurefile-csi` storage class that comes out of the box in openshift, the worker / master virtual machine must be able to talk to the cluster storage account.  

Set Network ACLs on the cluster storage account during Creation & AdminUpdate only if the Microsoft.Storage service endpoint is set on the subnets.  

### Test plan for issue:

There will be an e2e test that will run against production clusters to ensure this works based on the storage account controller.  

We can test this as follows though: 
- Disable the storage account controller
- Disable the service endpoints controller
- Disable `ensureServiceEndpoints` step in RP for dev clusters only
- Remove NACLs from cluster storage account
- Set service endpoints on one or all cluster subnets
- NACLs should only include subnets which have Microsoft.Storage service endpoints.  

### Is there any documentation that needs to be updated for this PR?

We need to publish something along the lines of:
- the default `azurefile-csi` storage class will not work if you don't have Microsoft.Storage service endpoints set on the subnets your OCP cluster uses.  The way around this is to create your own storage class.
